theia-extensions/viewer-prototype/src/browser/trace-explorer/trace-explorer-sub-widgets/trace-explorer-placeholder-widget.tsx

```diff
@@ -95,18 +95,22 @@ export class TraceExplorerPlaceholderWidget extends ReactWidget {
             }).then(async progress => {
                 progress.report({ message: 'Launching trace server... ', work: { done: 10, total: 100 } });
                 try {
-                    const resolve = await this.traceServerConfigService.startTraceServer(this.path, this.port);
+                    const resolve = await this.traceServerConfigService.startTraceServer(this.path, this.args);
                     if (resolve === 'success') {
-                        progress.report({ message: `Trace server started on port: ${this.port}.`, work: { done: 100, total: 100 } });
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        const port = this.args!.match(/(?<=port=)\d+/g)![0];
```

Comment:
I wonder if we can get rid of using the port at all, since it still makes an assumption about the structure of the CLI args. Instead we can output something like:
"Trace server started using the following arguments: ..."

If we still decide to keep the port, can we move that Regex to something like PreferenceService, to avoid repeating it in other locations?

@bhufmann , any thoughts?

Reply:
> @bhufmann , any thoughts?

I'll get back to you later on today.

Reply:
Here I think we shouldn't mention the port anymore. I agree with the suggestion of @mirsky-work.